### PR TITLE
Add KLUE-TC (Korean Language Understing Evaluation - Topic Classification) dataset

### PR DESCRIPTION
**Pull request by taeminlee**
## Checklist for adding MMTEB dataset

Reason for dataset addition:
KLUE-TC is a quality dataset annotated by human for topic classification task of news titles.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

points
- dataset 2 + new language in classication 4 = 6

### CODE REVIEW

File: docs/mmteb/points/605.jsonl [hunk 1]
```diff
@@ -0,0 +1 @@
+{"GitHub": "taeminlee", "New dataset": 6}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/605.jsonl:1]: add points for review

File: mteb/tasks/Classification/kor/KlueTC.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        text_creation="found",
+        bibtex_citation="""@misc{park2021klue,
+      title={KLUE: Korean Language Understanding Evaluation}, 
+      author={Sungjoon Park and Jihyung Moon and Sungdong Kim and Won Ik Cho and Jiyoon Han and Jangwon Park and Chisung Song and Junseong Kim and Yongsook Song and Taehwan Oh and Joohong Lee and Juhyun Oh and Sungwon Lyu and Younghoon Jeong and Inkwon Lee and Sangwoo Seo and Dongjun Lee and Hyunwoo Kim and Myeonghwa Lee and Seongbo Jang and Seungwon Do and Sunkyoung Kim and Kyungtae Lim and Jongwon Lee and Kyumin Park and Jamin Shin and Seonghyun Kim and Lucy Park and Alice Oh and Jungwoo Ha and Kyunghyun Cho},
+      year={2021},
+      eprint={2105.09680},
+      archivePrefix={arXiv},
+      primaryClass={cs.CL}
+}""",
+        n_samples={"validation": 9107},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/kor/KlueTC.py:39]: I would reduce to 2048 samples

File: mteb/tasks/Classification/kor/KlueTC.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+from mteb.abstasks import AbsTaskClassification
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class KlueTC(AbsTaskClassification):
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/kor/KlueTC.py:7]: Any reason why this is not a clustering task?

File: docs/mmteb/points/605.jsonl (hunk 1 above)
Comment (taeminlee) [reply docs/mmteb/points/605.jsonl:1]: yep i'll add points

File: mteb/tasks/Classification/kor/KlueTC.py (hunk 2 above)
Comment (taeminlee) [reply mteb/tasks/Classification/kor/KlueTC.py:39]: OK, that's not a problem

File: mteb/tasks/Classification/kor/KlueTC.py (hunk 3 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/Classification/kor/KlueTC.py:7]: Let us keep it in line with the paper then to allow results to be at least semi-comparable

File: mteb/tasks/Classification/kor/KlueTC.py (hunk 2 above)
Comment (KennethEnevoldsen) [reply mteb/tasks/Classification/kor/KlueTC.py:39]: please do use the stratified subsampling (suggested above)

Comment (taeminlee) [reply mteb/tasks/Classification/kor/KlueTC.py:39]: I've use stratified subsampling, and run models on the subset. :)

**Comment by taeminlee** [comment]
@KennethEnevoldsen I think all issues have been resolved, next time I submit, I'll do subsampling beforehand. Please merge. Thank you. :)

File: mteb/tasks/Classification/kor/KlueTC.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        text_creation="found",
+        bibtex_citation="""@misc{park2021klue,
+      title={KLUE: Korean Language Understanding Evaluation}, 
+      author={Sungjoon Park and Jihyung Moon and Sungdong Kim and Won Ik Cho and Jiyoon Han and Jangwon Park and Chisung Song and Junseong Kim and Yongsook Song and Taehwan Oh and Joohong Lee and Juhyun Oh and Sungwon Lyu and Younghoon Jeong and Inkwon Lee and Sangwoo Seo and Dongjun Lee and Hyunwoo Kim and Myeonghwa Lee and Seongbo Jang and Seungwon Do and Sunkyoung Kim and Kyungtae Lim and Jongwon Lee and Kyumin Park and Jamin Shin and Seonghyun Kim and Lucy Park and Alice Oh and Jungwoo Ha and Kyunghyun Cho},
+      year={2021},
+      eprint={2105.09680},
+      archivePrefix={arXiv},
+      primaryClass={cs.CL}
+}""",
+        n_samples={"validation": 9107},
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/kor/KlueTC.py:39]:
```suggestion
        n_samples={"validation": 2048},
```
Assumed

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good. Added a minor fix and enabled auto-merge

File: mteb/tasks/Classification/kor/KlueTC.py (hunk 4 above)
Comment (taeminlee) [reply mteb/tasks/Classification/kor/KlueTC.py]: Aha, thank you. 2048 That's right.

**Comment by taeminlee** [comment]
OMG. I'm not very good with github, and something went wrong during the sync, and the PR got closed. Luckily I have some existing work as a branch, so I'll reopen the PR, I'm really sorry.

**Pull request closed without merging**